### ISSUE TITLE
TypeError: no implicit conversion of Integer into String

### ISSUE THREAD

**Issue by groupzer0** [logstash-plugins/logstash-output-nagios_nsca#15]
I am having a problem with logstash-output-nagios_nsca and the latest version of Logstash 7.1.1

Centos7

Steps to reproduce: 

setup config:
 
```
    input {
    beats {
	  port => 5074 
          type => "nagioscheckbeats"
	  ssl => true
	  ssl_certificate_authorities => ["/etc/logstash/certs/cacert.pem"]
	  ssl_certificate => "/etc/logstash/certs/{{ inventory_hostname }}.crt"
	  ssl_key => "/etc/logstash/certs/{{ inventory_hostname }}-logstash.pem"
	  ssl_verify_mode => "force_peer"
      }
    }

    filter {
		if [type] =~ "nagiosmetric" {
			mutate {
			  convert => { "name" => "string" }
			  convert => { "status_code" => "string" }
			  convert => { "message" => "string" }
			}
		}
	}
	
output {

		if [type] == "nagiosmetric" {
		  elasticsearch { 
			hosts => [{{ logstash_elasticsearch_hosts }}]
			user => {{ xpack_management_elasticsearch_username }}
			password => {{ xpack_management_elasticsearch_password }}
			cacert => "{{ logstash_ssl_dir }}/{{ logstash_ssl_certificate_file }}"
			ssl_certificate_verification => false
			index => "nagioscheckbeat-%{+YYYY.MM.dd}" 
		}
		} else {
		    nagios_nsca {
				host => "{{nagios_server}}"
				port => 5667
				nagios_service => "%{name}"
				nagios_status => "%{[status_code]}"
				nagios_host => "%{[beat][hostname]}"
				message_format => "%{name}: %{message}"
			}
			# used for debugging
			#file {
			#  	path => "/tmp/logstash-output.txt"
			#  	codec => line { format => "custom format: %{message}"}
			#}
		}
	}
```

The debug output is:

```
[2019-08-31T02:12:26,725][DEBUG][logstash.outputs.nagiosnsca] Running send_nsca command     {:nagios_nsca_command=>"/usr/sbin/send_nsca -H xxxx.xxxx.com -p 5667 -d :", :message=>"xxxx.xxxx.com:heartbeat:0:heartbeat: OK: Hello<br/>"}

[2019-08-31T02:12:26,727][WARN ][logstash.outputs.nagiosnsca] Skipping nagios_nsca output; error calling send_nsca {:error=>#<TypeError: no implicit conversion of Integer into String>, :nagio
s_nsca_command=>"/usr/sbin/send_nsca -H xxxx.xxxx.com -p 5667 -d :", :message=>"xxxx.xxxx.com:heartbeat:0:heartbeat: OK: Hello<br/>", :missed_event=>#<LogStash::Event:0x669acbab>}   
[2019-08-31T02:12:26,727][DEBUG][logstash.outputs.nagiosnsca] Backtrace {:backtrace=>["org/jruby/RubyProcess.java:1567:in `spawn'", "org/jruby/RubyKernel.java:1646:in `spawn'", "uri:classload
er:/META-INF/jruby.home/lib/ruby/stdlib/open3.rb:202:in `popen_run'", "uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/open3.rb:98:in `popen3'", "/usr/share/logstash/vendor/bundle/jruby/
2.5.0/gems/logstash-output-nagios_nsca-3.0.6/lib/logstash/outputs/nagios_nsca.rb:135:in `send_to_nagios'", "/usr/share/logstash/vendor/bundle/jruby/2.5.0/gems/logstash-output-nagios_nsca-3.0.
6/lib/logstash/outputs/nagios_nsca.rb:110:in `receive'", "/usr/share/logstash/logstash-core/lib/logstash/outputs/base.rb:89:in `block in multi_receive'", "org/jruby/RubyArray.java:1792:in `ea
ch'", "/usr/share/logstash/logstash-core/lib/logstash/outputs/base.rb:89:in `multi_receive'", "org/logstash/config/ir/compiler/OutputStrategyExt.java:118:in `multi_receive'", "org/logstash/co
nfig/ir/compiler/AbstractOutputDelegatorExt.java:101:in `multi_receive'", "/usr/share/logstash/logstash-core/lib/logstash/java_pipeline.rb:235:in `block in start_workers'"]}`
```

I know next to nothing about Ruby, but I messed around with the plugin code enough to narrow the problem down to the cmd variable was trying to join strings with the port as an integer.

I modified the code enough to be able to see the messages coming through to Nagios. The below is NOT meant to represent an improvement on the plugin code - it's just a hack for me to locate the problem and prove that the message can get through. Hopefully, someone else can provide the correct solution in ruby.

I changed this:

```
def cmd
    return @cmd if @cmd
   # @cmd = [@send_nsca_bin, "-H", @host, "-p", @port, "-d", ":"]
   # @cmd = @cmd + ["-c", @send_nsca_config]  if @send_nsca_config
   @cmd = @send_nsca_bin.to_s + " " + "-H" + " " + @host.to_s + " " + "-p" + " " + @port.to_s + " " + "-d" + " " + ":"
  # @cmd = @cmd + [" " + "-c" + " " + @send_nsca_config  if @send_nsca_config
  @cmd
end

def send_to_nagios(cmd, message)
  Open3.popen3(*cmd) do |i, o, e|
    # added logger for testing
    logger.info("Running send_nsca command", :nagios_nsca_command => cmd, :message => message)
    i.puts(message)
    i.close
   end
end
```

These changes remove the error and allow data to be processed as expected. 

Thanks,

**Comment by AdminValida** [comment]
We are facing the same problem using Logstash 7.14.1 and nagios_nsca plugin 3.0.6.

I changed:

`config :port, :validate => :number, :default => 5667`

to

`config :port, :validate => :string, :default => "5667"`

in nagios_nsca.rb.

Regards

**Comment by andsel** [comment]
The version `3.0.6` gem contains:
```ruby
@cmd = [@send_nsca_bin, "-H", @host, "-p", @port, "-d", ":"]
```

but this was fixed with https://github.com/logstash-plugins/logstash-output-nagios_nsca/commit/7d83ab5081559429f4d19e4eaca15223305689a0#diff-7d65683094e75031404a4fa5a4821da15518945c312dfc75ed65c74cd09e47b2 and never published

**Comment by roaksoax** [comment]
Just to updated that the fix has been included in version 3.0.7, which is now widely available.
